File: lab6-openmp/lab6-openmp_f90/omp_bug1.f90
! -*- Mode: Fortran90; -*-
!=================================================================

program Bug1
  !-----------------------------------------------------------------
  ! FILE: omp_bug1.f90
  ! DESCRIPTION:
  !   This example attempts to show use of the PARALLEL DO 
  !   construct.  However it will generate errors at compile time.  
  !   Try to determine what is causing the error.  See 
  !   omp_bug1fix.f90 for a corrected version.
  ! AUTHOR: Blaise Barney  5/99
  ! Upgraded to F90, free-source form: Daniel R. Reynolds, 2/2009
  !-----------------------------------------------------------------
  !======= Inclusions ===========

  !======= Declarations =========
  implicit none

  integer :: TID, i, chunk
  integer, external :: omp_get_thread_num
  integer, parameter :: N = 50
  integer, parameter :: chunksize = 5
  real :: A(N), B(N), C(N)

  !======= Internals ============

  !     Some initializations
  do i = 1, N
     A(i) = i * 1.0
     B(i) = A(i)
  enddo
  chunk = chunksize

  !$omp parallel do shared(A,B,C,chunk) private(i,TID) &
  !$omp schedule(static,chunk)
  TID = omp_get_thread_num()
  do i = 1, N
     C(i) = A(i) + B(i)
     print *,'TID= ',TID,'i= ',i,'C(i)= ',C(i)
  enddo
  !$omp end parallel do

end program Bug1
!=================================================================

File: lab6-openmp/lab6-openmp_cxx/omp_bug1.cpp
/* FILE: omp_bug1.cpp
   DESCRIPTION:
   This example attempts to show use of the parallel for construct.  
   However it will generate errors at compile time.  Try to determine 
   what is causing the error.  See omp_bug1fix.cpp for a corrected 
   version.
   AUTHOR: Blaise Barney  5/99
   UPDATED: Daniel R. Reynolds (updated to C++), 1/13/2013 */

#include <omp.h>
#include <stdio.h>
#include <stdlib.h>
#define N           50
#define CHUNKSIZE   5

int main (int argc, char *argv[]) {

  // local variables
  int i, chunk, tid;
  float a[N], b[N], c[N];

  // Some initializations
  for (i=0; i < N; i++)  a[i] = b[i] = i * 1.0;
  chunk = CHUNKSIZE;

# pragma omp parallel for			\
  shared(a,b,c,chunk)				\
  private(i,tid)				\
  schedule(static,chunk)
  {
    tid = omp_get_thread_num();
    for (i=0; i<N; i++) {
      c[i] = a[i] + b[i];
      printf("tid = %i, i = %i, c[i]= %g\n", tid, i, c[i]);
    }
  }  // end parallel for

  return 0;
} // end main

